oauth_generic.go

```diff
@@ -104,12 +108,28 @@ func (c genericOauthClient) inspectOauthAccessToken(ctx context.Context, accessT
 		return nil, errors.New("unable to inspect access token")
 	}
 
-	var inspectResponse InspectResponse
-	if err := limitedJsonUnmarshal(resp.Body, infoRequestMaxLen, &inspectResponse); err != nil {
+	// since we don't know what the JSON from the server will look like, we create a
+	// generic interface and then map manually to values set in the config
+	var genericInterface interface{}
```

Comment:
Maybe use a `map[string]interface{}` as the type here, instead of an `interface{}`? That should reduce some code and get rid of the following `m` variable below.

Reply:
Done in b262fa144c995d206c02aa4ed378c9d0b234c77a